From 0d7790b1cd9b0013ffdba957961def87b75bf99c Mon Sep 17 00:00:00 2001
From: Yongjun Zhang <yzhang@cloudera.com>
Date: Fri, 4 Mar 2016 13:25:29 -0800
Subject: [PATCH 1304/2848] HDFS-9889. Update balancer/mover document about
 HDFS-6133 feature. Contributed by Yongjun Zhang.

(cherry picked from commit 8e08861a14cb5b6adce338543d7da08e9926ad46)
(cherry picked from commit fb6661f3bdda95f17e15dbc52d1c89faf082946e)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/HDFSCommands.md

Change-Id: I16c1e106b0fe66d0e117964275d11687a2f8ca45
---
 .../hadoop-hdfs/src/site/apt/HDFSCommands.apt.vm   |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/site/apt/HDFSCommands.apt.vm b/hadoop-hdfs-project/hadoop-hdfs/src/site/apt/HDFSCommands.apt.vm
index 3b5fccc..47ce88a 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/site/apt/HDFSCommands.apt.vm
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/site/apt/HDFSCommands.apt.vm
@@ -169,6 +169,8 @@ HDFS Commands Guide
    Note that the <<<blockpool>>> policy is more strict than the <<<datanode>>>
    policy.
 
+   Besides the above command options, a pinning feature is introduced starting from 2.7.0 to prevent certain replicas from getting moved by balancer/mover. This pinning feature is disabled by default, and can be enabled by configuration property "dfs.datanode.block-pinning.enabled". When enabled, this feature only affects blocks that are written to favored nodes specified in the create() call. This feature is useful when we want to maintain the data locality, for applications such as HBase regionserver.
+
 ** <<<datanode>>>
 
    Runs a HDFS datanode.
@@ -384,7 +386,9 @@ HDFS Commands Guide
 | -f \<local file\>  | Specify a local file containing a list of HDFS files/dirs to migrate.
 *--------------------+--------------------------------------------------------+
 
-  Note that, when both -p and -f options are omitted, the default path is the root directory.
+   Note that, when both -p and -f options are omitted, the default path is the root directory.
+
+   In addition, a pinning feature is introduced starting from 2.7.0 to prevent certain replicas from getting moved by balancer/mover. This pinning feature is disabled by default, and can be enabled by configuration property "dfs.datanode.block-pinning.enabled". When enabled, this feature only affects blocks that are written to favored nodes specified in the create() call. This feature is useful when we want to maintain the data locality, for applications such as HBase regionserver.
 
 ** <<<namenode>>>
 
-- 
1.7.9.5

